### PR TITLE
feat: user subscriptions + subscription endpoints impl

### LINKED ISSUE #1692: [notifications] Subscription management API
# Description

Implement POST/DELETE /v1/notifications/subscriptions; user-level only for MVP.
The subscribe and unsubscribe actions should update Brevo lists

### PR DESCRIPTION
**Pull request by cka-y**
**Summary:**
This pull request introduces a complete implementation of notification subscription management for both authenticated users and unauthenticated (public) APIs, including integration with the Brevo email provider for announcement subscriptions. It adds new API endpoints, shared helpers, and comprehensive unit tests, enabling users to manage their notification subscriptions and ensuring synchronization with the external email provider.

Key changes include:

**API and Endpoint Additions**
* Added new public and authenticated subscription management endpoints by implementing `SubscriptionsApiImpl` and fully implementing subscription methods in `UsersApiImpl`, allowing users to create, update, list, and delete notification subscriptions. [[1]](diffhunk://#diff-90666583a5a3b7081018dcf063eeb65fd2c391a2e5d6b7d5e8e12a1ab7e79302R1-R55) [[2]](diffhunk://#diff-546587c1fa1bcfa15eae403cfc8a297ac19085239fbb4ca011a7fd216963efb3L122-R223)
* Registered the new `subscriptions_api` routes with the main application to expose these endpoints. [[1]](diffhunk://#diff-7fb84fb8cda14524d6f7782b2352f56bd8aa7c41eb2cfca574d6d092e7ff871dR63-R64) [[2]](diffhunk://#diff-c03edaf8ac6a8151e382b7dfd02a323226733c4c4c000af2eb658ca692ccd793R30) [[3]](diffhunk://#diff-c03edaf8ac6a8151e382b7dfd02a323226733c4c4c000af2eb658ca692ccd793R63)

**Brevo Integration**
* Refactored and expanded Brevo (email provider) integration: extracted contact/list management into reusable helpers, enabled syncing of `api.announcements` subscriptions with Brevo, and handled provider errors gracefully. [[1]](diffhunk://#diff-999957f919d8b7d7a3ee11f71a0f3b54bbb22e50096bfe62920d53b9151752b2R43-R90) [[2]](diffhunk://#diff-999957f919d8b7d7a3ee11f71a0f3b54bbb22e50096bfe62920d53b9151752b2L59-R107) [[3]](diffhunk://#diff-917d6a44112a67eb37f458bd9bf08575d812f468b8a9c1cbad29dfd4aa115e9aR1-R39)

**Database and Model Layer**
* Added `NotificationSubscriptionImpl` for converting ORM subscription objects to Pydantic models, supporting both API layers.

**Testing**
* Added comprehensive unit tests for the new public subscriptions API, covering success, error, and edge cases, including Brevo integration and error handling.
* Updated test imports to support mocking Brevo helpers.

These changes provide robust, production-ready notification subscription management with external provider synchronization and thorough test coverage.
Summarize the changes in the pull request including how it relates to any issues (include the #number, or link them).

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [x] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: docs/UserServiceAPI.yaml [hunk 1]
```diff
@@ -222,6 +224,56 @@ paths:
         "501":
           description: Not yet implemented.
 
+  /v1/subscriptions/{id}:
+    get:
+      summary: Get a subscription by ID (unauthenticated)
```
Comment (davidgamez) [inline docs/UserServiceAPI.yaml]: This endpoint is, in fact, protected by authentication/authorization.

File: api/src/shared/common/brevo.py [hunk 2]
```diff
@@ -40,6 +40,54 @@ class BrevoSubscriptionStatus(Enum):
...
+    Uses create_contact with update_enabled so it works whether or not the
+    contact already exists.
+    """
+    api = _get_contacts_api()
+    api.create_contact(
+        sib_api_v3_sdk.CreateContact(
+            email=email,
+            attributes={"MDB_SUBSCRIPTION_ID": subscription_id},
+            list_ids=[list_id],
+            update_enabled=True,
```
Comment (davidgamez) [inline api/src/shared/common/brevo.py:73]: 👍

File: api/src/shared/common/brevo.py [hunk 3]
```diff
@@ -40,6 +40,54 @@ class BrevoSubscriptionStatus(Enum):
...
+
+    Uses create_contact with update_enabled so it works whether or not the
+    contact already exists.
+    """
+    api = _get_contacts_api()
+    api.create_contact(
+        sib_api_v3_sdk.CreateContact(
+            email=email,
+            attributes={"MDB_SUBSCRIPTION_ID": subscription_id},
+            list_ids=[list_id],
```
Comment (davidgamez) [inline api/src/shared/common/brevo.py:72]: [question]: Does the `update_enabled` add the list_id to the lists, or does it `reset` the list to only one list?

File: docs/UserServiceAPI.yaml [hunk 4]
```diff
@@ -222,6 +224,56 @@ paths:
...
+        "200":
+          description: Subscription retrieved.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/NotificationSubscription"
+        "404":
+          description: Subscription not found.
+    delete:
+      summary: Delete a subscription by ID (unauthenticated)
```
Comment (davidgamez) [inline docs/UserServiceAPI.yaml]: The same here; this endpoint is authenticated, even though it's not an "user" endpoint

File: api/src/shared/common/brevo.py (hunk 3 above)
Comment (cka-y) [reply api/src/shared/common/brevo.py:72]: In the doc the field `list_ids` is described as `Ids of the lists to add the contact to` so I believe it only adds. Also according to my tests with my own account, i haven't been removed from any other list when calling this endpoint.

File: docs/UserServiceAPI.yaml (hunk 4 above)
Comment (cka-y) [reply docs/UserServiceAPI.yaml]: fixed ✅

File: docs/UserServiceAPI.yaml (hunk 1 above)
Comment (cka-y) [reply docs/UserServiceAPI.yaml]: fixed ✅

**Review by davidgamez — APPROVED** [review]
